core_lib/src/interface/scribblearea.cpp

```diff
@@ -1478,15 +1478,34 @@ void ScribbleArea::setLayerVisibility(LayerVisibility visibility)
     updateAllFrames();
 }
 
+void ScribbleArea::updateLayerVisibility()
+{
+    switch (mLayerVisibility) {
+    case LayerVisibility::ALL:
+        mEditor->preference()->set(SETTING::LAYER_VISIBILITY, 0);
+        break;
+    case LayerVisibility::RELATED:
+        mEditor->preference()->set(SETTING::LAYER_VISIBILITY, 1);
+        break;
+    case LayerVisibility::CURRENTONLY:
+        mEditor->preference()->set(SETTING::LAYER_VISIBILITY, 2);
+        break;
+    default:
+        break;
+    }
+}
```

Comment:
There's no need for a switch here, simply static cast the enum to int and set preference using that.. eg..
mPrefs->set(SETTING::LAYER_VISIBILITY, static_cast<int>(mLayerVisibility));

Also notice, we already hold a reference to PreferenceManager, so there's no need to call editor to get it.

Reply:
Done. Thanks for the advice.